test/imath/ref/out.xml

```diff
@@ -0,0 +1,517 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<TranslationUnit name="/home/anders/code/cppmm/test/imath/bind/imath_box.cpp" id="5">
+    <Record name="Imath_3_0::Box&lt;Imath_3_0::Vec3&lt;float&gt; &gt;" id="6" width="192" align="32">
```

Comment:
is width the size of the type ?

Reply:
Yes. it's what clang calls it internally. I can change that to "size" in the output. Might change it internally since I keep writing "align" as "height" because of muscle memory...

Reply:
Incidentally did you know that std::string is 32 bits in gcc and 24 bits in clang? I didn’t until yesterday when I thought I had a bug in the size generation. 

Clang seems to use gcc’s libstdc++ during binding generation, at least on my system, I assume because that’s what I have setup as my “default” compiler. I don’t know enough about Linux toolchains to know what that means for other people though. If you somehow managed to generate an opaquebytes representation of string using libc++, then tried to use it in a libstdc++ environment, that would obviously be very bad.

Reply:
yes, I did know that stl type sizes change between stl variants and stl versions. The famous one being when g++ moved from a reference counted string (which I think was 8 bytes, the size of a pointer) to the short string optimisation I think in g++5 (which was 24 bytes).  libc++ is 24 bytes as well, because I think it was the libc++ team that initially went with SSO. If you go back to the original discussion we had about memory layout I mention this.
https://github.com/vfx-rs/rfcs/pull/1
I also mention the itanium standard for abi layout and the size change between the hardware you're on, 32bit machine or 64 bit machine.

But for me it's a cost vs benefit decision like the other ways of wrapping. If the headers are being generated as part of the build process it isn't an issue, but depending on the type and the scope of use, I'd expect you'd choose the write way to wrap it...